libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -4426,6 +4461,11 @@ toPredsNumSpec v (NumSpecInterval ml mu) =
     [assert $ Lit l <=. v | l <- maybeToList ml]
       ++ [assert $ v <=. Lit u | u <- maybeToList mu]
 
+numSpecHasError :: forall fn a. (Ord a, HasSpec fn a) => NumSpec fn a -> Maybe (NE.NonEmpty String)
+numSpecHasError s@(NumSpecInterval (Just a) (Just b))
+  | a > b = Just ("NumSpec has low bound greater than hi bound" :| ["   " ++ show s])
+numSpecHasError _ = Nothing
```

Comment:
Doesn't `guardNumSpec` already do something like that?

Reply:
They are very similar, but there types are a bit different
```
guardNumSpec ::
  (Ord n, HasSpec fn n, TypeSpec fn n ~ NumSpec fn n) => [String] -> NumSpec fn n -> Specification fn n
 
numSpecHasError :: forall fn a. (Ord a, HasSpec fn a) => NumSpec fn a -> Maybe (NE.NonEmpty String)
```